src/components/App/ExpressRecorder.tsx

```diff
@@ -356,21 +367,36 @@ export class ExpressRecorder extends Component<ExpressRecorderProps, State> {
             item.stop();
         });
 
-        this.setState({ doUpload: true });
+        this.setState({
+            doUpload: true
+        });
     };
 
     /**
      * triggered when recording is finished
      * @param recordedBlobs
+     * @param duration
+     * @param screenBlobs
      */
-    handleRecordingEnd = (recordedBlobs: Blob[], duration: number) => {
+    handleRecordingEnd = (
+        recordedBlobs: Blob[],
+        duration: number,
+        screenBlobs: Blob[] | undefined
+    ) => {
         const blob = new Blob(recordedBlobs, { type: "video/webm" });
         const self = this;
         // handle chrome blob duration issue
         fixWebmDuration(blob, duration, function(fixedBlob: Blob) {
             self.setState({ recordedBlobs: recordedBlobs, blob: fixedBlob });
             self.dispatcher.dispatchEvent(RecorderEvents.recordingEnded);
         });
+        if (screenBlobs) {
+            const screenBlobObject = new Blob(screenBlobs, { type: "video/webm" });
+            const self = this;
+            fixWebmDuration(screenBlobObject, duration, function(fixedBlob: Blob) {
+                self.setState({ screenRecordedBlobs: screenBlobs, screenRecordedBlob: fixedBlob });
+            });
+        }
```

Comment:
I'm not a big fan of the `const self = this`.
I would have created a proxy private function that takes the same arguments and pass them to the relevant util function.
something like:
```
_proxyFixWebmDuration = (blob: Blob, duration: number, target: Blob, newStateParams: Partial<State>) => {
  fixWebmDuration(blob, duration, (fixedBlob: Blob) => {
    this.setState({
      ...this.state,
      [target]: fixedBlob,
      ...newStateParams
    });
  });
}
```

and use it like:
```
if(screenBlobs){
  this._proxyFixWebmDuration(new Blob(screenBlobs, { type: "video/webm" }), duration, 'screenRecordedBlob', { screenRecordedBlobs: screenBlobs});
}
```

with this approach, you can avoid the `self` variable and re-use it anywhere from the component.

Reply:
thanks, I copy-paste it so I didn't notice we don't need self here if we use arrow function.